File: perl/problem2.pl
#my $n1;
#my $n2 = 1;
#my $sum = 2;
#my $evens = 2;
#while ($sum < 4000000){
#	$n1 = $n2;
#	$n2 = $sum;
	#$sum = $n1 + $n2;
	#if (!($sum % 2)){
	#	$evens += $sum;
	#}
#}

my ($num, $last) = (1,1);
my $sum = 0;
while ($num < 4000000){
	($num, $last) = ($num + $last, $num);
	$sum += $num unless $num % 2;
}
print $sum;

File: rust/problem2.rs
fn main() {
    let mut a = 1;
    let mut b = 1;
    let mut sum = 0;
    while b <= 4000000 {
        if b % 2 == 0 {
            sum += b
        }

        let tmp = a;
        a = b; 
        b = tmp + b;
    }
    println!("{}", sum)
}

